apps/system/js/battery_overlay.js

```diff
@@ -48,21 +48,22 @@
 
     _start: function bm_init() {
       this.getAllElements();
-      var battery = this._battery;
-      if (battery) {
+      return navigator.getBattery().then((battery) => {
+        this._battery = battery;
+
         // When the device is booted, check if the battery is drained. If so,
         // batteryshutdown would be triggered to inform sleepMenu shutdown.
-        this.checkBatteryDrainage.bind(this);
+        this.checkBatteryDrainage();
 
         battery.addEventListener('levelchange', this);
         battery.addEventListener('chargingchange', this);
-      }
-      window.addEventListener('screenchange', this);
+        window.addEventListener('screenchange', this);
 
-      this._screenOn = true;
-      this._wasEmptyBatteryNotificationDisplayed = false;
+        this._screenOn = true;
+        this._wasEmptyBatteryNotificationDisplayed = false;
```

Comment:
nah, it's just github diff getting confused. I intended the whole block because it's inside a getBattery promise callback now.